doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,314 @@
+.. _nested_virt:
+
+Enable Nested Virtualization
+############################
+
+With nested virtualization enabled in ACRN, you can run virtual machine
+instances inside of a guest VM (also called a user VM) running on the ACRN hypervisor.
+Although both "level 1" guest VMs and nested guest VMs can be launched
+from the Service VM, the following distinction is worth noting:
+
+* The VMX feature (``CPUID01.01H:ECX[5]``) does not need to be visible to the Service VM
+  in order to launch guest VMs. A guest VM not running on top of the
+  Service VM is considered a level 1 (L1) guest.
+
+* The VMX feature must be visible to an L1 guest to launch a nested VM. An instance
+  of a guest hypervisor (KVM) runs on the L1 guest and works with the
+  L0 ACRN hypervisor to run the nested VM.
+
+The conventional single-level virtualization has two levels - the L0 host
+(ACRN hypervisor) and the L1 guest VMs. With nested virtualization enabled,
+ACRN can run guest VMs with their associated virtual machines that define a
+third level:
+
+* The host (ACRN hypervisor), which we call the L0 hypervisor
+* The guest hypervisor (KVM), which we call the L1 hypervisor
+* The nested guest VMs, which we call the L2 guest VMs
+
+.. figure:: images/nvmx_1.png
+   :width: 700px
+   :align: center
+
+   Generic Nested Virtualization
+
+
+High Level ACRN Nested Virtualization Design
+********************************************
+
+The high-level design of nested virtualization in ACRN is shown in :numref:`nested_virt_hld`.
+Nested VMX is enabled by allowing a guest VM to use VMX instructions,
+and emulating them using the single level of VMX available in the hardware.
+
+A logical processor uses Virtual-Machine Control Structures (VMCSs) while
+it is in a VMX operation. The core of the NVMX emulation is that ACRN builds VMCS02
```

Comment:
Would you add an explanation of the ``VMCS01``, ``VMCS02``, and ``VMCS12`` names meanings.  I presume all VMs have a VM control structure?  Also, the figure below has nothing labeled as the "L0 hypervisor" or "L2 hypervisor" mentioned in the text in the figure, Should the L2 hypervisor be L1 hypervisor (or maybe L2 VM)?

The text bullets in the image should be pulled out and added instead as bullets in the text so it could be easily edited and searched.

Reply:
There is no "L2 hypervisor", will fix the typo and add "L0 hypervisor" to the figure (:name: nested_virt_hld). The concept of VMCS01, VMCS02, and VMCS12 are not very straightforward, and I can add more explanation in the HLD doc.

Reply:
> Would you add an explanation of the `VMCS01`, `VMCS02`, and `VMCS12` names meanings. I presume all VMs have a VM control structure? Also, the figure below has nothing labeled as the "L0 hypervisor" or "L2 hypervisor" mentioned in the text in the figure, Should the L2 hypervisor be L1 hypervisor (or maybe L2 VM)?
> 
> The text bullets in the image should be pulled out and added instead as bullets in the .rst file so it could be easily edited and searched.

Yes, good idea!